lib/core/collection/array.nit

```diff
@@ -301,6 +301,30 @@ abstract class AbstractArray[E]
 	    self[a] = self[b]
 	    self[b] = e
 	end
+
+	# Rotates the elements of self once to the left
+	#
+	# ~~~nit
+	# var a = [12, 23, 34, 45]
+	# a.lrotate
+	# assert a == [23, 34, 45, 12]
+	# ~~~
+	fun lrotate do
+		var fst = shift
+		push fst
+	end
+
+	# Rotates the elements of self once to the right
+	#
+	# ~~~nit
+	# var a = [12, 23, 34, 45]
+	# a.rrotate
+	# assert a == [45, 12, 23, 34]
+	# ~~~
+	fun rrotate do
```

Comment:
can-you promote rotations to Sequence?

also, List and CircularArray shoud be a more efficient datastructure for roration. document this fact.